File: backend/flask_api/app/routes/booking.py
from flask import Blueprint, request, jsonify
from app.extensions import db
from sqlalchemy import text

booking_bp = Blueprint("booking", __name__)

@booking_bp.route("/", methods=["POST"])
def create_booking():
    data = request.json

    query = text("""
        INSERT INTO appointments_appointment
        (vehicle_id, service_id, staff_id, start_time, end_time, status, created_at)
        VALUES (:vehicle, :service, :staff, :start, :end, 'CONFIRMED', NOW())
    """)

    db.session.execute(query, {
        "vehicle": data["vehicle_id"],
        "service": data["service_id"],
        "staff": data["staff_id"],
        "start": data["start_time"],
        "end": data["end_time"],
    })

    db.session.commit()

    return jsonify(message="Appointment created"), 201

File: frontend/static/js/booking.js
document
.getElementById("bookingForm")
.addEventListener("submit", async (e)=>{

e.preventDefault()

const data={
vehicle_id:document.getElementById("vehicle").value,
service_id:document.getElementById("service").value,
staff_id:document.getElementById("staff").value,
start_time:document.getElementById("start").value,
end_time:document.getElementById("end").value
}

try{

await apiRequest("/booking","POST",data)

alert("Appointment created")

}catch(err){

alert("Error creating booking")

}

})